Source/WebCore/css/PropertySetCSSStyleDeclaration.cpp

```diff
@@ -317,16 +318,16 @@ String PropertySetCSSStyleDeclaration::getPropertyValueInternal(CSSPropertyID pr
     if (!propertyID || !isCSSPropertyExposed(propertyID))
         return { };
 
-    Document* doc = nullptr;
-    JSDOMObject* wrap = wrapper();
-    if (wrap) {
-        JSDOMGlobalObject* global = wrap->globalObject();
-        if (global) {
-            DOMWindow& window = activeDOMWindow(*global);
-            doc = window.document();
+    if (auto* wrapper = this->wrapper()) {
+        if (auto* globalObject = wrapper->globalObject()) {
+            if (auto* document = activeDOMWindow(*globalObject).document())
+                CSSPrimitiveValue::setUseLegacyPrecision(document->quirks().needsFlightAwareSerializationQuirk());
```

Comment:
Don't love this (or really any use of mutable globals these days if I am honest), too easy to become a problem down the line whether via unexpected side effects or threading. This is pretty targeted though.

Reply:
Given that CSS already uses a global for the value pool, I decided that I liked this better. The old code was really inconsistent about whether the document was passed through, and the document argument that can be null was not a very good mechanism. In a In an earlier version of the patch, I had fixed things to pass the document through more consistently, and then realized it was not really valuable given how we are trying to target this quirk. Note also that this is about implementing a quirk and tying parts of the system together that normally wouldn’t be in the name of figuring out whether to apply the quirk or not. I also considered performance and you’ll see that one of the slowest things here is the code wending its way all the way from a value to the document. Also, strangely, it relies on the wrapper to do so that it only works for values that have been exposed to JavaScript. On balance, I decided that I liked this for now, and I look forward to deleting it when the single website `flightaware.com` fixes its behavior.

Reply:
Still open to suggestions/advice on how to do it better though.

Reply:
Another way of thinking about it is that if we find a place to put the CSS value pool rather than a global, we can put this there too.

Maybe CSS _serialization_ should involve a serializer object, and we could move this flag there along with the `StringBuilder`. I think that’s the big winner idea.